libs/util/shared/src/lib/elasticsearch/elasticsearch.service.ts

```diff
@@ -42,41 +46,55 @@ export class ElasticsearchService {
         geometry
       ),
       _source: requestFields,
-      runtime_mappings: {
-        isSpatial: {
-          type: 'keyword',
-          script: `
-String result = 'no';
-String formats = doc.format.join('|').toLowerCase();
-if (formats.contains('geojson') || formats.contains('arcgis') || formats.contains('ogc') || formats.contains('shp')) result = 'yes';
-String protocols = doc.linkProtocol.join('|').toLowerCase();
-if (protocols.contains('esri') || protocols.contains('ogc')) result = 'yes';
-emit(result)
-`,
-        },
-        license: {
-          type: 'keyword',
-          script: `
-String raw = '';
-if (doc.containsKey('licenseObject.default.keyword') && doc['licenseObject.default.keyword'].length > 0)
-  raw += doc['licenseObject.default.keyword'].join('|').toLowerCase();
-if (doc.containsKey('MD_LegalConstraintsUseLimitationObject.default.keyword') && doc['MD_LegalConstraintsUseLimitationObject.default.keyword'].length > 0)
-  raw += doc['MD_LegalConstraintsUseLimitationObject.default.keyword'].join('|').toLowerCase();
+    }
+    this.processRuntimeFields(payload)
+    return payload
+  }
 
-if (raw.contains('odbl')) emit('ODbL');
-if (raw.contains('pddl')) emit('PDDL');
-if (raw.contains('odc-by')) emit('ODC-By');
-if (raw.contains('cc0') || raw.contains('cc-0')) emit('CC-0');
-if (raw.contains('cc-by') || raw.contains('cc by')) emit('CC BY');
-if (raw.contains('open license')) emit('Open License');
-if (raw.contains('etalab')) emit('Etalab');
-`,
-        },
-      },
+  // payload object will be mutated
+  private processRuntimeFields(payload: EsSearchParams): EsSearchParams {
+    const addMapping = (fieldName: string) => {
+      if (!payload.runtime_mappings) payload.runtime_mappings = {}
+      payload.runtime_mappings[fieldName] = {
+        type: 'keyword',
+        script: this.runtimeFields[fieldName],
+      }
+    }
+    const lookForField = (node: unknown) => {
+      if (typeof node === 'string' && node in this.runtimeFields) {
+        addMapping(node)
+        return
+      }
+      if (Array.isArray(node)) {
+        node.forEach(lookForField)
+        return
+      }
+      if (typeof node !== 'object') return
+      if ('field' in node && typeof node.field === 'string') {
+        if (node.field in this.runtimeFields) {
+          addMapping(node.field)
+        }
+      }
+      if ('query' in node && typeof node.query === 'string') {
+        for (const runtimeField in this.runtimeFields) {
+          if (node.query.indexOf(runtimeField + ':') > -1)
+            addMapping(runtimeField)
+        }
+      }
+      for (const key in node) {
+        if (typeof node[key] === 'object') lookForField(node[key])
+      }
     }
+    lookForField(payload.aggregations)
+    lookForField(payload.sort)
+    lookForField(payload.query)
```

Comment:
You can add `_source` I guess.

Reply:
I thought about it, it's actually possible to get the value of runtime fields but not through `_source`, only through the `fields` option which we don't use yet. This will be handy for the metadata quality level.